src/primitive_file_posix.cc

```diff
@@ -20,7 +20,7 @@
 #include "process.h"
 #include "objects_inline.h"
 
-#ifdef TOIT_POSIX
+#if defined(TOIT_POSIX) || defined(TOIT_FREERTOS)
```

Comment:
How bad would it be to simply split this and have a separate primitive_file_esp32.cc implementation? I'm not super excited about wrapping the calls in macros and I'd rather have a few helpers. Having said that, I can live with this solution for now :)

Reply:
There would be a lot of duplicate code, but otherwise there is nothing preventing this.
I do have a plan to look into the whole primitive_file complex and refactor it to use proper resources (that would give true async IO on windows for example). When that happens, the split would make sense.